api/src/main/java/jakarta/servlet/http/HttpSession.java

```diff
@@ -213,4 +219,24 @@ public interface HttpSession {
      */
     public boolean isNew();
 
+    /**
+     * Provides a mechanism for applications to interact with the {@code HttpSession} outside of the scope of an HTTP
+     * request.
+     * <p>
+     * To use this mechanism, applications call the {@code Consumer#accept(Object)} method on the {@code Consumer} instance
+     * returned from this method and pass an application provided instance of {@code Consumer<Session>}. During that call,
+     * the container will call the the {@code Consumer#accept(Session)} method of the application provided
+     * {@code Consumer<Session>} passing the {@code HttpSession} object.
+     * <p>
+     * For the purposes of session access, validity, passivation, activation etc. the container behaves as if the call the
+     * container makes to the {@code Consumer#accept(Session)} method of the application provided {@code Consumer<Session>}
+     * instance occurs during the processing of an HTTP request.
+     *
+     * @return A container provided {@code Consumer} instance that accepts application provided {@code Consumer<Session>}
+     * instances to enable the application to interact with the HTTP session outside of the scope of an HTTP request or
+     * {@code null} if access to the session is not supported outside of an HTTP request
+     */
+    default public Consumer<Consumer<HttpSession>> getAccessor() {
```

Comment:
I think we also need to allow this method to fail somehow, as some cluster based solutions might not want to allow access outside the scope of a HTTP request.

Reply:
Isn't that the null return?  Or do you want the returned consumer be able to fail at a later time?

Reply:
Sorry, I missed that, too early in the morning and had not had my coffee yet :-)